drv/gimlet-seq-server/src/main.rs

```diff
@@ -180,11 +217,10 @@ fn main() -> ! {
     // If the sequencer is already loaded and operational, the design loaded
     // into it should be willing to talk to us over SPI, and should be able to
     // serve up a recognizable ident code.
-    //
-    // TODO except for now we're going to skip the version check and
-    // unconditionally reprogram it because the SPI communication code ain't
-    // written, and also yolo. Replace this with a check.
-    let reprogram = true;
+    let seq = seq_spi::SequencerFpga::new(spi.device(SEQ_SPI_DEVICE));
+
+    let reprogram = if seq.valid_ident() { false } else { true };
```

Comment:
... should this be `!seq.valid_ident()`?

Reply:
Yes -- fixed